packages/grafbase-sdk/src/field/typedefs.ts

```diff
@@ -0,0 +1,298 @@
+import { RequireAtLeastOne } from 'type-fest'
+import { Enum } from '../enum'
+import {
+  BooleanListDef,
+  DateListDef,
+  ListDef,
+  NumberListDef,
+  StringListDef
+} from './list'
+import { ScalarType, Searchable } from '..'
+
+interface FieldLength {
+  min?: number
+  max?: number
+}
+
+export enum FieldType {
+  String = 'String',
+  Int = 'Int',
+  Email = 'Email',
+  ID = 'ID',
+  Float = 'Float',
+  Boolean = 'Boolean',
+  Date = 'Date',
+  DateTime = 'DateTime',
+  IPAddress = 'IPAddress',
+  Timestamp = 'Timestamp',
+  URL = 'URL',
+  JSON = 'JSON',
+  PhoneNumber = 'PhoneNumber'
+}
+
+export class SearchDef {
```

Comment:
What are these `Def` suffixes? Are they meant as "directive"?

Reply:
Definition.

Reply:
Sure, but why not just `Search`?

Reply:
Ok we can leave it as is